### PR TITLE
Added multiselect support

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
Addresses #23 and #13.

### CODE REVIEW

**Review by Blackbaud-ToddRoberts — DISMISSED** [review]

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]
LGTM once you get a passing build.

**Comment by Blackbaud-AlexKingman** [comment]
The build is failing because we need list-builder to merge first. The tests are missing a method.

I'll come back around and fix that once list-builder is merged.

File: e2e/list-view-grid.e2e-spec.ts [hunk 1]
```diff
@@ -40,8 +41,6 @@ describe('List view grid', () => {
   });
 
   it('should match the previous screenshot with column chooser open (screen: xs)', (done) => {
```
Comment (Blackbaud-TrevorBurch) [inline e2e/list-view-grid.e2e-spec.ts:43]: You removed the portion here that sets this test to the `xs` size.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts [hunk 2]
```diff
@@ -141,6 +150,21 @@ export class SkyListViewGridComponent
...
 
   public ngAfterContentInit() {
+
+    // Watch for selection changes and update _multiselectSelectedIds for local comparison.
+    this.state.map(s => s.selected.item)
+      .takeUntil(this.ngUnsubscribe)
+      .distinctUntilChanged(this.selectedMapEqual)
+      .subscribe((items: ListSelectedModel) => {
+        let selectedIds: string[] = [];
+        items.selectedIdMap.forEach((isSelected, id) => {
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/list-view-grid/list-view-grid.component.ts:162]: Personal preference here but I'd like to see a line break on each side of the `forEach` block as it is easier to read.

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
One important thing and one personal preference thing :)

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 2 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts:162]: yeah i think i can agree with that! updated.

File: e2e/list-view-grid.e2e-spec.ts (hunk 1 above)
Comment (Blackbaud-AlexKingman) [reply e2e/list-view-grid.e2e-spec.ts:43]: good catch! fixed.

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]
LGTM

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]
LGTM

File: package.json [hunk 3]
```diff
@@ -28,16 +28,16 @@
     "@skyux/grids": "^3.0.0",
     "@skyux/i18n": "^3.5.0",
     "@skyux/indicators": "^3.0.0",
-    "@skyux/list-builder": "^3.0.1",
-    "@skyux/list-builder-view-checklist": "^3.0.0",
+    "@skyux/list-builder": "^3.2.1",
+    "@skyux/list-builder-view-checklist": "^3.1.0",
```
Comment (Blackbaud-SteveBrush) [inline package.json]: Just confirming that both of these peer changes are required?

Comment (Blackbaud-AlexKingman) [reply package.json]: List builder - absolutely. It has all the nuts & bolts for multiselect to work. Checklist, on the other hand, wouldn't necessarily need updated here. I'll revert it back.

Comment (Blackbaud-AlexKingman) [reply package.json]: yep, it works fine. reverted.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts [hunk 4]
```diff
@@ -121,6 +128,8 @@ export class SkyListViewGridComponent
 
   public currentSearchText: Observable<string>;
 
+  private _multiselectSelectedIds: string[] = [];
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list-view-grid/list-view-grid.component.ts]: Is this member tied to a `get` or `set`? We usually reserve the underscore for getter/setter values.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts [hunk 5]
```diff
@@ -141,6 +150,23 @@ export class SkyListViewGridComponent
   }
 
   public ngAfterContentInit() {
+
+    // Watch for selection changes and update _multiselectSelectedIds for local comparison.
+    this.state.map(s => s.selected.item)
+      .takeUntil(this.ngUnsubscribe)
+      .distinctUntilChanged(this.selectedMapEqual)
+      .subscribe((items: ListSelectedModel) => {
+        let selectedIds: string[] = [];
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list-view-grid/list-view-grid.component.ts]: `selectedIds` value never changes; can be `const`.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts [hunk 6]
```diff
@@ -232,13 +258,38 @@ export class SkyListViewGridComponent
...
+      this.dispatcher.toolbarShowMultiselectToolbar(true);
+    }
   }
 
   public ngOnDestroy() {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
   }
 
+  public multiselectSelectionChange(event: any): void {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list-view-grid/list-view-grid.component.ts]: Is this literally an `any` type, or is there a specific type that could be injected here?

File: src/app/public/modules/list-view-grid/list-view-grid.component.html [hunk 7]
```diff
@@ -1,19 +1,25 @@
...
     [hasToolbar]="hasToolbar | async"
+    [height]="height | async"
     [highlightText]="highlightSearchText ? (currentSearchText | async) : undefined"
+    [multiselectRowId]="multiselectRowId"
     [rowHighlightedId]="rowHighlightedId"
-    (selectedColumnIdsChange)="columnIdsChanged($event)"
+    [selectedColumnIds]="selectedColumnIds | async"
     [sortField]="sortField | async"
+    [width]="width | async"
+    (multiselectSelectionChange)="multiselectSelectionChange($event)"
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list-view-grid/list-view-grid.component.html]: Would it be possible to prefix the method with `on`, (e.g., `onMultiselectSelectionChange`)?

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts [hunk 8]
```diff
@@ -388,6 +439,30 @@ export class SkyListViewGridComponent
     return true;
   }
 
+  private selectedMapEqual(prev: ListSelectedModel, next: ListSelectedModel): boolean {
+    if (prev.selectedIdMap.size !== next.selectedIdMap.size) {
+      return false;
+    }
+
+    for (let i = 0; i < next.selectedIdMap.size; i++) {
+      const key = Array.from(next.selectedIdMap.keys())[i];
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/list-view-grid/list-view-grid.component.ts]: I could be wrong, but it appears `Array.from` is not supported by IE 11.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Good stuff! Mostly code style suggestions.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 5 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: You sure? I'm pushing new values to it in the following `.foreach()`

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 4 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: got it. fixed.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 6 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: nope, its `SkyGridSelectedRowsModelChange`. fixed.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 8 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: Doh, good catch. Maps are tricky and there's only a limited amount of ways to iterate through them. I restructured this a bit to get around using `Array.from`.

File: src/app/public/modules/list-view-grid/list-view-grid.component.html (hunk 7 above)
Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.html]: yep, done.

File: src/app/public/modules/list-view-grid/list-view-grid.component.ts (hunk 5 above)
Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: The contents of the array change, but the array itself is never reassigned.

Comment (Blackbaud-AlexKingman) [reply src/app/public/modules/list-view-grid/list-view-grid.component.ts]: Well I learned something today!
https://stackoverflow.com/questions/23436437/why-can-i-change-value-of-a-constant-in-javascript

Fixed.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**